pkg/machines/components/create-vm-dialog/createVmDialog.jsx

```diff
@@ -358,6 +407,49 @@ class CreateVM extends React.Component {
                 </label>
                 {installationSourceVal}
 
+                <hr />
+
+                { this.state.sourceType != EXISTING_DISK_IMAGE_SOURCE &&
+                <React.Fragment>
+                    <label className="control-label" htmlFor="storage-pool-select">
+                        {_("Storage")}
+                    </label>
+                    <Select.Select id="storage-pool-select"
+                                   initial={this.state.storagePool}
+                                   onChange={e => this.onChangedValue('storagePool', e)}>
+                        <Select.SelectEntry data="NewVolume" key="NewVolume">{"Create New Volume"}</Select.SelectEntry>
+                        <Select.SelectDivider />
+                        <optgroup key="Storage Pools" label="Storage Pools">
+                            { this.props.storagePools.map(pool => <Select.SelectEntry data={pool.name} key={pool.name}>{pool.name}</Select.SelectEntry>)}
+                        </optgroup>
+                    </Select.Select>
+
+                    { this.state.storagePool !== "NewVolume" &&
+                    <React.Fragment>
+                        <label className="control-label" htmlFor="storage-volume-select">
+                            {_("Volume")}
+                        </label>
+                        <Select.Select id="storage-volume-select"
+                                       initial={this.state.storageVolume}
+                                       onChange={e => this.onChangedValue('storageVolume', e)}>
+                            {volumeEntries}
+                        </Select.Select>
+
+                        { isVolumeUsed[this.state.storageVolume] && isVolumeUsed[this.state.storageVolume].length > 0 &&
+                        <HelpBlock>
+                            <p className="text-warning">{_("This volume is already used by another VM.")}</p>
+                        </HelpBlock> }
+                    </React.Fragment> }
+
+                    { this.state.storagePool === "NewVolume" &&
+                    <MemorySelectRow label={_("Size")}
```

Comment:
label is ignored in MemorySelectRow component now, you have to put it outside, otherwise label is missing.

Reply:
Needs:
```
<label htmlFor='storage-size' className='control-label'>         
  {_("Size")}                                               
  </label>                                                        
```